documentation/src/docs/controller-service.md

```diff
@@ -282,168 +277,193 @@ Controller service efficient queries for finding immediate successors
- on the stream. For example, during scaling the state of the stream
+ on the Stream. For example, during scaling the state of the Stream
  transitions from *active* to *scaling* and once scaling completes, it
- transitions back to *active*. Stream state is used as a barrier to
- ensure only one type of operation is being performed on a given stream
+ transitions back to *active*. Stream State is used as a barrier to
+ ensure only one type of operation is being performed on a given Stream
  at any point in time. Only certain state transitions are allowed and
- are described in the State Transition object. Only legitimate state
+ are described in the state transition object. Only legitimate state
  transitions are allowed and any attempt for disallowed transition
  results in appropriate exception.
 
 #### Truncation Record
- This corresponds to the stream cut which was last used to truncate the
- given stream. All stream segment queries superimpose the truncation
- record and return segments that are strictly greater than or equal to
- the stream cut in truncation record.
+ This corresponds to the `StreamCut` which was last used to truncate the
+ given Stream. All Stream Segment queries superimpose the truncation
+ record and return Stream Segments that are strictly greater than or equal to
+ the `StreamCut` in truncation record.
+
+#### Sealed Segments Maps
+Once the Stream Segments are sealed, controller needs to store additional information about the Stream Segment. Presently, we have two types of information:
+
+ - Epoch, the Stream Segment was sealed in.
+ - Size of the Stream Segment at the time of sealing.
+
+ These records have two different characteristics and are used in different types of queries. For example;
+
+ 1. Sealing epoch is important for querying _successor Stream Segments_.(For each Stream Segment we store its sealing epoch directly in the metadata store.)
+ 2. Stream Segment sizes are used during truncation workflows.(For sealed sizes, we store it in a map of Segment to size at the time of sealing.)
+ 3. Successor queries are performed on a single Stream Segment whereas truncation workflows work on a group of Stream Segments.
+
+ This ensures that during truncation we are able to retrieve sealed sizes for multiple Stream Segments with minimal number of calls into underlying metadata store. Since we could have arbitrarily large number of Stream Segments that have been sealed away, we cannot store all of the information in a single map and hence we shard the map and store it. The sharding function we use is to hash the creation epoch and get the shard number.
 
 #### Sealed Segments Record
- Since the segment table is append only, any additional information
- that we need to persist when a segment is sealed is stored in sealed
- segments record. Presently, it simple contains a map of segment number
+ Since the Segment Table is append only, any additional information
```

Comment:
@shivesh  can i chnage Segment Table to Segment record?

Reply:
in the write up i shared, i suggested removing this section (`Sealed Segments Record`) and replacing it with
```#### Sealed Segments Map
Once segments are sealed, controller needs to store additional information about the segment. Presently we have two types of information that we store namely the epoch it was sealed in and the overall size of the segment at the time of sealing. 
These records have two different characteristics and are used in different types of queries - for example, sealing epoch is important for querying successor segments; segment sizes are used during truncation workflows. Successor queries are performed on a single segment whereas truncation workflows work on a group of segments.
So we maintain two kinds of records for these two types of records. For each segment we store its sealing epoch directly in the metadata store. For sealed sizes, we store it in a map of segment to size at the time of sealing. This ensures that during truncation we are able to retrieve sealed sizes for multiple segments with minimal number of calls into underlying metadata store.
Since we could have arbitrarily large number of segments that have been sealed away, 
we cannot store all of the information in a single map and hence we shard the map and store it. The sharding function we use is to hash the creation epoch and get the shard number. 

```

Reply:
Please use the new text in the document i shared with you and dont mix and match it with existing document. 
i removed all the stale sections and replaced them with relevant sections.

Reply:
Yes! i checked and done!